### PR TITLE
Insert service name for authentication

### PR DESCRIPTION
**Pull request by anuptalwalkar**
GFM-277

### CODE REVIEW

File: modules/uhttp/client/client.go [hunk 1]
```diff
@@ -27,18 +27,22 @@ import (
 	"strings"
 
 	"go.uber.org/fx"
+	"go.uber.org/fx/config"
 
 	"golang.org/x/net/context/ctxhttp"
 )
 
+var serviceName string
```
Comment (alsamylkin) [inline modules/uhttp/client/client.go]: Nit: I thought we are using _ prefix for globals?

File: modules/uhttp/client/client_test.go [hunk 2]
```diff
@@ -27,15 +27,20 @@ import (
...
+applicationID: test
+`)
+
 var _defaultHTTPClient = &http.Client{Timeout: 2 * time.Second}
-var _defaultUHTTPClient = New(_defaultHTTPClient)
+var _defaultUHTTPClient = New(config.NewYAMLProviderFromBytes(testYaml), _defaultHTTPClient)
 
 func TestNew(t *testing.T) {
-	uhttpClient := New(_defaultHTTPClient)
+	uhttpClient := New(config.NewYAMLProviderFromBytes(testYaml), _defaultHTTPClient)
```
Comment (alsamylkin) [inline modules/uhttp/client/client_test.go]: "Document" panic in a test with a config without applicationID?:)

File: auth/localauth.go [hunk 3]
```diff
@@ -40,6 +40,9 @@ func defaultAuth() Client {
 
 // Instance returns initialized instance of auth client
 func Instance() Client {
+	if _std == nil {
```
Comment (alsamylkin) [inline auth/localauth.go]: do we care about thread safety here?

Comment (anuptalwalkar) [inline auth/localauth.go]: Good point. removing this check actually. I actually don't need default instance here, and I don't want lock in the request path.

File: modules/uhttp/client/client_filters.go [hunk 4]
```diff
@@ -86,6 +86,10 @@ func authenticationFilter(
 	ctx fx.Context, req *http.Request, next BasicClient,
 ) (resp *http.Response, err error) {
 	authClient := auth.Instance()
+
+	// Client needs to know what service it is to authenticate
+	authClient.SetAttribute(ctx, auth.ServiceAuth, _serviceName)
```
Comment (alsamylkin) [inline modules/uhttp/client/client_filters.go]: Do we need to set it up for every request? It doesn't seem to change..

Comment (anuptalwalkar) [inline modules/uhttp/client/client_filters.go]: Yes, we need to set it every time on the context in order for authenticate to work. (ctx is new per request)

**Pull request merged**